Spider Infestation Balance Rebalance Expansion Part 1

Hoolny opened the pull request:
## About The Pull Request
https://hackmd.io/dxhPOOdRRKW7siqPbpfmYQ?view The HackMD goes in-depth into what this PR is about

## Content
### Ambush Spider
![image](https://user-images.githubusercontent.com/84478872/236697303-16fb5fee-7862-457f-a053-9e634873f205.png)
![image](https://user-images.githubusercontent.com/84478872/236697338-7908e9ba-07d2-4af6-9e0a-583e74d96634.png)
### Scout Spider
![image](https://user-images.githubusercontent.com/84478872/236697485-91560e01-365d-4a9e-8401-a52b412d865f.png)
![image](https://user-images.githubusercontent.com/84478872/236697489-a9f1d46d-1e65-4e9d-995e-335e4cf40f8c.png)
![dreamseeker_JtR1iVptrV](https://user-images.githubusercontent.com/84478872/236697538-4e9ac1a4-b2b7-463a-b4ff-75f8ae2d266a.png)
## Tangle
![image](https://github.com/tgstation/tgstation/assets/84478872/756de834-5470-4a2a-9e8e-55d6a3ca770d)
![TkK4NHc](https://github.com/tgstation/tgstation/assets/84478872/4fa15fa6-db11-46a5-9b56-8e5fce4bc1c0)
![XxcLU0z](https://github.com/tgstation/tgstation/assets/84478872/7a007a94-e9a4-4700-9a0e-f5e2a178c924)

### Spiderlings
![ruCweIc](https://user-images.githubusercontent.com/84478872/236697721-b96469df-e68b-47b7-9316-ec152629e9aa.png)
![SWE0bgo](https://user-images.githubusercontent.com/84478872/236697726-7753f363-99db-46b1-8181-0dc755c1804e.png)
![BYNLUQa](https://user-images.githubusercontent.com/84478872/236697753-48c1c5b0-064a-45e9-bf19-ad2e29cdb426.png)


## Why It's Good For The Game
Currently, spiders have plenty of issues that make them a very 2d mentional antag they kill,die, and respawn which is boring and leads to a bunch of balance issues for their current design this PR both fixes the current issues spiders have and adds more depth into the antag.
## Changelog
:cl:
add: Added 3 new spider types
add: Added more spider abilities
balance: Rebalanced spiders
fix: Fixes spiderlings having density
/:cl:

Hoolny committed:
SCOUT SPIDER VERSION 1

Adds a new type of support spider the scout spider with the ability to alert of intruders and scout around the station

Hoolny committed:
Update spider_variants.dm

Hoolny committed:
Other spider scout additions

Hoolny committed:
I will not die today

Hoolny committed:
never again

Hoolny committed:
for now

Hoolny committed:
FUCK

Hoolny committed:
Update spider_variants.dm

Hoolny committed:
yes

Hoolny committed:
god is death

Hoolny committed:
no oh no

Hoolny committed:
damn

Hoolny committed:
IT WORKS BITCHES

Hoolny committed:
add ambush

Hoolny committed:
fixed

Hoolny committed:
spider buffs

Hoolny committed:
tangle spider

Hoolny committed:
t

Hoolny committed:
pain fix

Hoolny committed:
maybe it works

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
diff

Hoolny committed:
ye

Hoolny committed:
yes

Hoolny committed:
yes

Hoolny committed:
yey

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
chaps

Hoolny committed:
robobitch

Hoolny committed:
balance/rebalance/rebalance balance

Hoolny committed:
yes

Hoolny committed:
yep

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
yes

Hoolny committed:
spiderlings to spiderlings

Hoolny committed:
yes spiderlign sprites back

Hoolny committed:
spiderlings to the high seas

Hoolny committed:
yes

Hoolny committed:
Fixes Infinite Solid Web Glitch

Hoolny committed:
some little fixes over there and over here

Hoolny committed:
tangle coder sprites

Hoolny committed:
ye

Hoolny committed:
Update web_spikes.dm

Hoolny committed:
ye

Hoolny committed:
tangle webs

Hoolny committed:
yes

Hoolny committed:
some touches

Hoolny committed:
yep

ZephyrTFA commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -101,6 +101,62 @@
+	max_integrity = 40
+	alpha = 200
+	plane = GAME_PLANE_UPPER
+	var/can_tangle = TRUE
```
where is this used

SethLafuente replied:
thats for web_passage game_plane_upper is to make it go above mobs alpha makes it somewhat transparent integrity is for structure health and can_tangle gives it the properties to be able to tagle carbon that arent spiders

optimumtact replied:
I can't find a reference to can tangle anywhere else in this code

ZephyrTFA commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -101,6 +101,62 @@
+
+/obj/structure/spider/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/atmos_sensitive, mapload)
```
you make it atmos sensitive, and then do nothing atmos related

ZephyrTFA commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -128,3 +184,34 @@
+	else if(isliving(mover))
+		if(istype(mover.pulledby, /mob/living/basic/spiderling))
+			return TRUE
+		if(prob(100))
```
why even have a prob here

SethLafuente replied:
not anymore

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/communication_spiders.dm:
```diff
@@ -0,0 +1,42 @@
+/datum/action/communication_spiders/proc/spider_communication(mob/living/user, message)
+	if(!message)
+		return
+	var/my_message = span_spider("<>Command from [user]:</> [message]")
```
Whats up with `<>` and `</>`

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/solid_web.dm:
```diff
@@ -0,0 +1,64 @@
+		owner?.balloon_alert(owner, "interrupted!") // Null check because we might have been interrupted via being disintegrated
+	build_all_button_icons()
+
+	/// Creates a web in the current turf
```
```suggestion
/// Creates a web in the current turf
```

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/sticky_web.dm:
```diff
@@ -0,0 +1,63 @@
+		owner?.balloon_alert(owner, "interrupted!") // Null check because we might have been interrupted via being disintegrated
+	build_all_button_icons()
+
+	/// Creates a web in the current turf
```
```suggestion
/// Creates a web in the current turf
```

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm:
```diff
@@ -62,7 +62,7 @@
 	new /obj/structure/spider/stickyweb(target_turf)
 
-/// Variant for genetics, created webs only allow the creator passage
+/// Variant for genetics, created webs only allow the creator snare
```
This makes me think you did a blind replace all.

SethLafuente replied:
I did indeed

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_passage.dm:
```diff
@@ -0,0 +1,64 @@
+		owner?.balloon_alert(owner, "interrupted!") // Null check because we might have been interrupted via being disintegrated
+	build_all_button_icons()
+
+	/// Creates a web in the current turf
```
```suggestion
/// Creates a web in the current turf
```

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_spikes.dm:
```diff
@@ -0,0 +1,64 @@
+		owner?.balloon_alert(owner, "interrupted!") // Null check because we might have been interrupted via being disintegrated
+	build_all_button_icons()
+
+	/// Creates a web in the current turf
```
```suggestion
/// Creates a web in the current turf
```

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling.dm:
```diff
@@ -24,7 +25,7 @@
 	melee_damage_upper = 4
 
-	health = 5 // very low.
+	health = 5 // very low, so true besti
```
No.

SethLafuente replied:
:( no bestie

ZephyrTFA commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling_subtypes.dm:
```diff
@@ -3,23 +3,82 @@
+	desc = "Furry and brown, it makes you shudder to look at it. This one has sparkling red eyes."
+	icon_state = "guard_spiderling"
+	icon_dead = "guard_spiderling_dead"
+	desc = "Furry and brown, specializing in defense of the hive and other spides."
```
This is probably supposed to be `menu_description`

SethLafuente replied:
yeah it is

Hoolny committed:
small fixes

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_spikes.dm

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

Hoolny committed:
ye

Hoolny committed:
Update spiderwebs.dm

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
fixed it

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/spider_abilities/solid_web.dm

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

Hoolny committed:
woops to many spaces

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
GENEREALIZED sneak

Hoolny committed:
grammar error

Hoolny committed:
ye

SethLafuente commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_passage.dm:
```diff
@@ -0,0 +1,64 @@
+/// Make a solid web under yourself for area fortification
+/datum/action/cooldown/web_passage
+	name = "Spin Web Passage"
+	desc = "Spin a web passage to hide the nest from prey view."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "lay_web_passage"
```
dosen't that replace the original web with it

SethLafuente commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/communication_spiders.dm:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Sends a message to all currently living spiders.
+ */
+/datum/action/communication_spiders
+	name = "Communication"
+	desc = "Send a command to all living spiders."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
```
good point I can put it in hivemind

SethLafuente replied:
done did it just need to make it not be in loud letters figuring that out

Hoolny committed:
OOG

SethLafuente commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_sneak.dm:
```diff
@@ -0,0 +1,32 @@
+/datum/action/cooldown/web_sneak
+	name = "Sneak"
+	desc = "Blend into the shadows to stalk your prey."
```
made the SNEAK_ALIEN_TRAIT into SNEAK_TRAIT as the note said to probably do

KittyNoodle commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -1,3 +1,32 @@
+	melee_damage_upper = 30
+	speed = 5
+	player_speed_modifier = -3.1
+	menu_description = "Slow spider variant specializing in stalking and ambushing prey, but has goo health and damage."
```
goo health

KittyNoodle reviewed (commented):
didn't mean to open review

Wallemations commented:
Can we see some videos of how these new spiders function?

Hoolny committed:
goo health

SethLafuente commented:
> Can we see some videos of how these new spiders function?

glady I do have to say the walking animation of two of the spiders are in the works

Hoolny committed:
might not work

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
buncha fixes for Mr.melbert sir

Hoolny committed:
whaat light chat and dark chat have different things

Jackraxxus commented:
The lack of seething balanceposters commenting on this PR shows how few people actually read design docs.
Either that or the change is objectively perfect, if that's the case then congrats OP.

SethLafuente commented:
> The lack of seething balanceposters commenting on this PR shows how few people actually read design docs. Either that or the change is objectively perfect, if that's the case then congrats OP.

I am great at balancing I would say so myself as I do play the game a lot and I understand spider dynamics very well but for real Im planning on this being a test merge so I can see how this actually plays out you can't really fully tell if something is balanced if you haven't seen how it actually plays out and I'm sure there is a bunch of unbalanced things.

Also that's probably because maintainers are to busy looking at my absolute horrible coding

Hoolny committed:
made structure differences

Jackraxxus commented:
I like the look of things. The focus on webs and the growth time in particular. Only slightly worried people will complain even more about spiders being xenos 2 electric boogaloo. Also like that it incentivizes playing Broodmother, will be better than watching 10 ghosts circle the egg cluster waiting for someone else to take it so they can have all the fun murderboning.

SethLafuente commented:
> Can we see some videos of how these new spiders function?

Also sadly enough videos are to big so pictures will do

Hoolny committed:
woops white spot

SethLafuente commented:
> I like the look of things. The focus on webs and the growth time in particular. Only slightly worried people will complain even more about spiders being xenos 2 electric boogaloo. Also like that it incentivizes playing Broodmother, will be better than watching 10 ghosts circle the egg cluster waiting for someone else to take it so they can have all the fun murderboning.

fore sure broodmother now can make 2 web trap types and 2 new web protection types I should put that in the images as well also their egg laying time has being reduced and their health has being increased to 200 half of xeno queen and 2/3 of space dragon so is up to match to its importance that with a bunch of different changes that changes the spider dynamic broodmother will take more of a lead controlling role over the hive it won't be like xenos at all I promise

Hoolny committed:
var/override_message

IndieanaJones commented:
I'm interested in seeing a testmerge of this to get a feel for how it plays. There's nothing in this design document that stands out to me as bad, however, I'm just curious how it looks when all the puzzle pieces are put together.

nukularpower commented:
Like like the look of this, but since it already seems to be taking inspiration, I gotta ask - why not just port over/steal Paradise terror spiders, which are already proven to work well as a very dangerous midround/antag

Hoolny committed:
lets see if thsi works

SethLafuente commented:
> like the look of this, but since it already seems to be taking inspiration, I gotta ask - why not just port over/steal the design of Paradise terror spiders entirely, which are already proven to work well as a very dangerous midround/antag



> like the look of this, but since it already seems to be taking inspiration, I gotta ask - why not just port over/steal the design of Paradise terror spiders entirely, which are already proven to work well as a very dangerous midround/antag

I asked maintainers they told me it was a bad idea, I still looked at the code it was a lot more different and I saw reusing xeno traits and other to be an easier alternative and although terror spiders are a great antag very much they come with some flaws in their design and structure that doesn't synergies well with TG so I decided to make ones that go well with the TG gameplay, although this is based on the terror spider theme and how it works on station its very much extremely different

Hoolny committed:
stickyweb

Hoolny committed:
ye

Hoolny committed:
woops silly me

Hoolny committed:
ye

SethLafuente commented:
Well this is just rude >( fuck you linter I think all the code is working but linter is fucked rn

pretty much a finish on 2 spider sprites and this is done

Hoolny committed:
mhvbm

Hoolny committed:
Update code/game/objects/effects/spiderwebs.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
,m

Hoolny committed:
ye

Hoolny committed:
fix probably

Hoolny committed:
fix of previous issue (not my fault)

Hoolny committed:
simpler than it seemed

Hoolny committed:
health lower

Hoolny committed:
god please work (I know it works >:(    )

Hoolny committed:
now it does

Hoolny committed:
FINALLY

Hoolny committed:
removes fluff text not neccesary anymore

Hoolny committed:
no points point point put put

Hoolny committed:
slight health buffs

Hoolny committed:
scout confusing solution

Hoolny committed:
hunters are a little bit faster

Hoolny committed:
BALANCE YES YES MORE OF THAT

Hoolny committed:
MORE BALANCE

ATH1909 commented:
how slow are these spiders outside of webs? since weaving a web requires a do_after(), aren't spiders basically fucked against ranged weaponry now?

Hoolny committed:
Tangle Spider and ambush spider sprite update

Hoolny committed:
yeey

Hoolny committed:
moving ambush

Hoolny committed:
yesss

Hoolny committed:
sprite upgrade

Hoolny committed:
MORE YES YES

SethLafuente commented:
> how slow are these spiders outside of webs? since weaving a web requires a do_after(), aren't spiders basically fucked against ranged weaponry now?
> 
> how are they supposed to get their first few kills against people who aren't standing next to a very obvious web mass when those people can easily outrun all kinds of spider, even when injured?

I decided to increase hunter web speed at the same exact speed of a human in a high mood

guard spiders like ambush and guard have received a big health boost letting them tank a decent bit of damage 

scout spiders can walk through vents find people to kill and such and spiders can combo and teamwork to get through these kinds of things plus more structure building meaning spiders can stand their ground more easily now

its hard to tell without testing how it will go in a round so Im going to want this to be a test merge to tweak a few things

Also yeah range weaponry does fuck them over and is meant to fuck them over but this comes more now too the spiders working as a team better than the crew

Hoolny committed:
CROIKEN

Hoolny committed:
woops forgot tangle spider self healing

SethLafuente commented:
Idk how it works but I know its usually done with this type of things

 I would like to test merge this and for the weight of spiders to be slightly increased so I can get on fixing any balance issues as soon as possible if you guys don't wanna increase the weight that fine I can wait for spiders to spawn I guess but I do want to test merge this

Hoolny committed:
Update traits.dm

yes siir

Hoolny committed:
one damn space bar

Hoolny committed:
better grammar

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
woops forgot a space

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
oh yeah is a subtype now

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
another subtype so no need for this

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

SethLafuente commented:
> Please make give a parent for this as requested, that both alien sneak and spider sneak share
> 
> ```dm
> /datum/action/cooldown/sneak
>     
> /datum/action/cooldown/sneak/spider
> 
> /datum/action/cooldown/speak/alien
> ```

so a new that has the basic files of sneak where should I put it in?

Hoolny committed:
yey

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

SethLafuente commented:
> Please make give a parent for this as requested, that both alien sneak and spider sneak share
> 
> ```dm
> /datum/action/cooldown/sneak
>     
> /datum/action/cooldown/sneak/spider
> 
> /datum/action/cooldown/speak/alien
> ```
I think it should be like that now?

Hoolny committed:
ye

SethLafuente commented:
> Please make this a subtype of `/datum/action/command_spiders/` as requested
> 
> ```
> /datum/action/command_spiders
>     var/message_span = "spiderbroodmother"
> 
> /datum/action/command_spiders/scout
>     message_span = "spiderscout"
> ```
> 
> etc.
> 
> If you don't know how look at `/datum/action/cooldown/spell/list_target/telepathy`. It does something very similar.

I tried doing this but it seems the command broodmother just keeps overriding the scout report regardless of what I do when I make it a subtype

Hoolny committed:
finally fixed thanks to MSO

SethLafuente commented:
> Still here

not anymore thanks to MSO

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

SethLafuente commented:
Mm not good

Rex9001 commented:
Very interested to see where this project goes, seems like great all-round changes!

Hoolny committed:
aaaaaaaaaaahAAAAAAH thanks melbert :)

Hoolny committed:
yep

Hoolny committed:
yep

Hoolny committed:
gives spiderlings no density as they should have

Hoolny committed:
yeep

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
small bal;ance

Hoolny committed:
AAAAAAAAAAAAH

Jacquerel commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -65,7 +65,7 @@
 		return TRUE
 	else if(isliving(mover))
-		if(istype(mover.pulledby, /mob/living/basic/giant_spider))
+		if(istype(mover.pulledby, /mob/living/basic/spiderling))
```
This doesn't seem right, spiders should be able to haul bodies through webs and conversely it would seem odd if spiderlings even _can_ do that.

Jacquerel commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -101,6 +101,38 @@
+/obj/structure/spider/solid
+	name = "solid web"
+	icon = 'icons/effects/effects.dmi'
+	desc = "A solid thick wall of web, airtight enough to block air flow."
```
```suggestion
	desc = "A solid wall of web, thick enough to block air flow."
```
Nitpick but you don't need both of these words here, they basically mean the same thing.
And same with "airtight enough to block airflow", that is already what airtight means.

Jacquerel commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -128,3 +160,31 @@
+/obj/structure/spider/sticky
+	name = "sticky web"
+	icon = 'icons/effects/effects.dmi'
+	desc = "extremly sticky soft silk ."
```
```suggestion
	desc = "Extremely soft and sticky silk."
```

Jacquerel commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -128,3 +160,31 @@
+
+/obj/structure/spider/sticky/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(isspider(mover))
+		return TRUE
+	else if(isliving(mover))
+		if(istype(mover.pulledby, /mob/living/basic/spiderling))
+			return TRUE
+		balloon_alert(mover, "stuck in web!")
+		return FALSE
```
```suggestion
	if(isspider(mover))
		return TRUE
	if(!isliving(mover))
		return
	if(isspider(mover.pulledby))
		return TRUE
	balloon_alert(mover, "stuck in web!")
	return FALSE
```

Few things here:
- You don't need an else following a return.
- You should "early return" whenever possible, basically meaning that if the rest of your proc would be indented inside an if statement then instead just invert the if statement and return from that point.
- You're checking if something is being dragged by a spiderling again?

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -1,3 +1,32 @@
+/**
+ * ### Spider Ambush
+ * A subtype of the giant spider which is faster, has toxin injection, but less health and damage.
```
This comment isn't true.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -18,6 +47,41 @@
+/mob/living/basic/giant_spider/hunter/Initialize(mapload)
+	. = ..()
+
+	AddElement(/datum/element/web_walker, /datum/movespeed_modifier/hunter_web)
```
Rather than add this individually to almost every single spider variant, make a var on `/mob/living/basic/giant_spider` which contains the movement speed modifier to apply and then run this code in the `Initialize` of the base class if it is not null.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -18,6 +47,41 @@
+	speed = 2.8
+	player_speed_modifier = -3.1
+	sight = SEE_SELF|SEE_MOBS
+	menu_description = "Fast spider variant specializing in scouting and alerting of prey ,with the ability to travel in vents."
```
```suggestion
	menu_description = "Fast spider variant specializing in scouting and alerting of prey, with the ability to travel in vents."
```

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -18,6 +47,41 @@
+
+/**
+ * ### Spider Scout
+ * A subtype of the giant spider which is faster, has phermone detection, but less health and damage.
```
Just say "can see mobs through walls" instead of pheromone detection so it is clear what you actually mean.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -55,6 +121,62 @@
+ * ### Tangle Spider
+ *
+ * A subtype of the giant spider which specializes in support skills.
+ * Nurses can place down webbing in a quarter of the time that other species can and can wrap other spiders' wounds, healing them.
```
Stop copying and pasting these comments without updating them.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -55,6 +121,62 @@
+	speed = 4
+	player_speed_modifier = -3.1
+	web_type = /datum/action/cooldown/lay_web/sealer
+	menu_description = "Support spider variant specializing in contruction to protect their brethren, but has very low amount of health and deals low damage."
```
May be worth noting that it injects acid, so its low melee damage is less relevant.

Jacquerel replied:
Also should mention here that they can heal themselves.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm:
```diff
@@ -55,6 +121,62 @@
+
+/mob/living/basic/giant_spider/tangle/Initialize(mapload)
+	. = ..()
+	var/datum/action/cooldown/lay_web/solid_web/web_solid = new(src)
+	web_solid.Grant(src)
+
+	var/datum/action/cooldown/lay_web/web_passage/passage_web = new(src)
+	passage_web.Grant(src)
+
+	var/datum/action/cooldown/lay_web/web_spikes/spikes_web = new(src)
+	spikes_web.Grant(src)
+
+	var/datum/action/cooldown/lay_web/sticky_web/web_sticky = new(src)
+	web_sticky.Grant(src)
```
All four of these abilities are always granted at the same time, I am wonder if there is some way of consolidating them into fewer buttons which does not compromise on UX... there might not be though.

SethLafuente replied:
> All four of these abilities are always granted at the same time, I am wonder if there is some way of consolidating them into fewer buttons which does not compromise on UX... there might not be though. 

I think modsuits have this I would like to implemented in the future when more spiders come but for now I'm not to sure how I would do that and it isn't a huge issue right now

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -51,7 +51,7 @@
 
 /**
- * Sends a message to all spiders from the target.
+ * Sends a big message to all spiders from the target.
```
You forgot to remove the "big" part when you refactored the ability, it's not true any more.

SethLafuente replied:
this one is for spider command which is still a big message

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -61,10 +61,36 @@
 		to_chat(ghost, "[link] [my_message]")
 	user.log_talk(message, LOG_SAY, tag = "spider command")
+
+/datum/action/command_spiders/proc/format_message(mob/living/user, message)
```
As it is a new proc, give this a documenting comment (preceded by `///` on the line above)

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -61,10 +61,36 @@
+ */
+/datum/action/command_spiders/communication_spiders
+	name = "Communication"
+	desc = "Send a command to all living spiders."
```
Is this true? I thought the point was that this one _wasn't_ a command

SethLafuente replied:
yeah thats just a typo

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -61,10 +61,36 @@
+/datum/action/command_spiders/communication_spiders
+	name = "Communication"
+	desc = "Send a command to all living spiders."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "message"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	check_flags = AB_CHECK_CONSCIOUS
```
```suggestion
	button_icon_state = "message"
```
Don't need to repeat properties from the supertype which are not changed.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -61,10 +61,36 @@
+	check_flags = AB_CHECK_CONSCIOUS
+
+/**
+ * Sends a message to all spiders from the target.
```
This comment is copy/pasted from a different proc, and isn't true.

SethLafuente replied:
I thought it sent a small message to all spiders its meant to do that does it not?

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm:
```diff
@@ -83,3 +83,60 @@
+	webbing_time = 5 SECONDS
+
+/datum/action/cooldown/lay_web/solid_web/obstructed_by_other_web()
+	return !!(locate(/obj/structure/spider/solid) in get_turf(owner))
```
Is this true? Do you want solid web to be able to be placed on top of non-solid web?
Do you want non-solid webs to be able to be placed on top of solid webs? You are not accounting for either.

SethLafuente replied:
> Is this true? Do you want solid web to be able to be placed on top of non-solid web?
> 
> Do you want non-solid webs to be able to be placed on top of solid webs? You are not accounting for either.

That's intended for spiders to be able to hide traps and other webs below existing solid webs much like when you put girders and walls above tiles

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm:
```diff
@@ -83,3 +83,60 @@
+	webbing_time = 4 SECONDS
+
+/datum/action/cooldown/lay_web/web_passage/obstructed_by_other_web()
+	return !!(locate(/obj/structure/spider/passage) in get_turf(owner))
```
Should this be able to be placed on top of other solid or non-solid webs? Currently you are only preventing a passage from being overlaid onto another passage.
This is also true for all the following subtypes, so I'm not going to repeat it two more times.

It's starting to look like these structures should all share a subtype for ease of use.

SethLafuente replied:
> Should this be able to be placed on top of other solid or non-solid webs? Currently you are only preventing a passage from being overlaid onto another passage.
> 
> This is also true for all the following subtypes, so I'm not going to repeat it two more times.
> 
> 
> 
> It's starting to look like these structures should all share a subtype for ease of use.

Passage is specifically designed to have spider spikes, normal webs and sticky webs below it putting a web passage and a solid web together is essentially just reinforcing a web which for the current change in going for with spiders being more of s defend the castle kind of antag im all about giving them the freedom to do so it also looks nice

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm:
```diff
@@ -83,3 +83,73 @@
+	new /obj/structure/spider/passage(target_turf)
+
+
+/datum/action/cooldown/lay_web/sticky_web
+	name = "Spin Sticky Web"
+	desc = "Spin a web to stick intruders in place."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "lay_sticky_web"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED
+	cooldown_time = 20 SECONDS
+	webbing_time = 3 SECONDS
```
You resolved this but didn't do it.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_sneak.dm:
```diff
@@ -0,0 +1,32 @@
+/datum/action/cooldown/sneak/spider
```
Where is the `/datum/action/cooldown/sneak` from which this and alien sneak inherit?
It looks like you have got halfway to making these share behaviour but didn't actually implement the supertype, so these are both still copy/pasted abilities rather than sharing code.

Jacquerel replied:
Yes you want to make an action `/datum/action/cooldown/sneak` in whatever seems like an appropriate location.
All of the code shared by both of the abilities should be in there, these two extended versions should only have any code which is different between the two versions.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling.dm:
```diff
@@ -10,20 +10,22 @@
 	desc = "It never stays still for long."
 	icon_state = "spiderling"
 	icon_dead = "spiderling_dead"
+	density = FALSE
```
If you did this in a separate PR already then doing it here too means you will conflict with yourself when it is merged.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling.dm:
```diff
@@ -10,20 +10,22 @@
 	initial_language_holder = /datum/language_holder/spider
 	basic_mob_flags = FLAMMABLE_MOB | DEL_ON_DEATH
 	mob_size = MOB_SIZE_TINY
+	var/menu_description = "Normal spiderling."
```
This should have a documenting comment.
However, as I advised you before, I won't merge this while the menu displays spiderlings. The egg selection menu should display the grown spider people are choosing, not the spiderling.

SethLafuente replied:
I personally disagree I think it's better to show people what they will actually become other than what they will become so players understand they will be spawning as spiderlings the menu description gives the abilities and how the spider looks is purely cosmetic 

It's an egg it produces spiderlings not fully grown spiders

Jacquerel replied:
Yes players should see what they are actually becoming: the adult spider. This is useful information, the spiderling is not useful information.
Unless you can convince another maintainer otherwise I will DNM this until you fix this.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling_subtypes.dm:
```diff
@@ -3,23 +3,81 @@
+	desc = "Furry and brown, it makes you shudder to look at it. This one has sparkling red eyes."
+	icon_state = "guard_spiderling"
+	icon_dead = "guard_spiderling_dead"
+	menu_description = "Furry and brown, specializing in defense of the hive and other spides."
```
This is a bad pattern, we should get this data from the grown spider and don't need it on the spiderling. Having to write the same information in two places is not good.

SethLafuente replied:
That's what Sans did I'm just building on it by adding more spiderlings

Jacquerel replied:
San did not do this because they did not have a menu description until you added it.

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling.dm:
```diff
@@ -62,7 +64,7 @@
 	AddComponent(\
 		/datum/component/growth_and_differentiation,\
-		growth_time = 10 MINUTES,\
+		growth_time = 1 MINUTES,\
```
This may have a significant effect on botany spiderlings, perhaps ones created from weeds need their own subtype which grows more slowly.

SethLafuente replied:
botany spiderlings taking 10 minutes to grow is a bit ridiculous in all honesty 1 minute seems more appropriate

Jacquerel commented on code/modules/mob/living/basic/space_fauna/spider/spiderlings/spiderling_subtypes.dm:
```diff
@@ -3,23 +3,81 @@
 /mob/living/basic/spiderling/guard
 	grow_as = /mob/living/basic/giant_spider
+	name = "guard spiderling"
+	desc = "Furry and brown, it makes you shudder to look at it. This one has sparkling red eyes."
```
If you are going to give these all individual descriptions, they shouldn't be copy/pasted from the adult spider because it does not make sense for them to have the same description.
Either write unique descs or don't have unique descs.

SethLafuente replied:
This is why I have an issue with giant_spider being the exact same code and implementation as guard spider... well you see they have the same description because they are the same spider whoever coded the spiders decided to make one of the spiders from the spider infestation the guard spider the giant_spider I told maintainers about it and they said they didn't see this as an issue so I just ignored it

Jacquerel replied:
This has absolutely no relevance to my review comment.

Jacquerel commented on code/modules/mob_spawn/ghost_roles/spider_roles.dm:
```diff
@@ -93,7 +96,7 @@
 
 /obj/effect/mob_spawn/ghost_role/spider/process(seconds_per_tick)
-	amount_grown += rand(0, 1) * seconds_per_tick
+	amount_grown += rand(0, 10) * seconds_per_tick
```
This introduces a large degree of variance, are you sure that is what you want?

Jacquerel replied:
It doesn't gain nothing or 10, it gains anywhere between 0 and 10.
It could gain 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, or 10.
This makes the time it will take much more random than it was previously, it could still take as long as it did previously.

SethLafuente replied:
MM im not sure if this randomness is a good thing or a bad thing what do you think

SethLafuente replied:
eeh ill change it to 10-20

Jacquerel commented on code/modules/mob_spawn/ghost_roles/spider_roles.dm:
```diff
@@ -75,9 +75,12 @@
+		/mob/living/basic/spiderling/ambush,
+		/mob/living/basic/spiderling/tangle,
+		/mob/living/basic/spiderling/guard,
+		/mob/living/basic/spiderling/scout,
```
As we spoke about in discord, you need to refactor the selection menu so that it displays the grown spider by reading from `spiderling:grow_as` instead of showing a radial menu of 6 nearly identical spiderlings differentiated only by colour.

SethLafuente replied:
![image](https://github.com/tgstation/tgstation/assets/84478872/1d612d5f-1d4f-4644-bd9e-8045da04138b)
I believe is a better design choice the spiderlings are both very easy to tell  a part show the info menu bar and show the name of what it is on the top screen

I think spawns should show you what you will spawn as an not what you will become is fully cosmetic anyways

I also prefer the look

Jacquerel replied:
You aren't choosing a spiderling because you want to play as a spiderling, you are choosing it for what it grows into. That is the important information.
The look is worse because it is less informative than showing you the fully grown spiders, which are different in more than colour.

This UX is significantly worse, and not acceptable.

SethLafuente replied:
I don't know what the UX is but fine

Jacquerel commented on code/modules/movespeed/modifiers/mobs.dm:
```diff
@@ -97,12 +97,27 @@
+/datum/movespeed_modifier/midwife_web
+	multiplicative_slowdown = 1.2
+
+/datum/movespeed_modifier/ambush_web
+	multiplicative_slowdown = 5
+
+/datum/movespeed_modifier/tarantula_web
+	multiplicative_slowdown = 5
```
If some of these have the same value, give them a more generic name and then just have one datum for that value. There's no reason to make multiple copies which do the same thing.

SethLafuente replied:
> If some of these have the same value, give them a more generic name and then just have one datum for that value. There's no reason to make multiple copies which do the same thing.

Mostly did it differently so if someone wanted to balance a spider they would have an easier time but I have no problem with merging them into let's say very, slow,slow,avarage and fast categories

Jacquerel reviewed (changes requested):
There's a few things here which seem to only be half-implemented, needs a bit of work yet before it is ready.

Hoolny committed:
Update spiderwebs.dm

Makes spiders be able to attack after placing long cooldown abilities

GolfinhoVoador commented:
Spiderlings currently get stuck in webs, lol

Hoolny committed:
Update code/game/objects/effects/spiderwebs.dm

Co-authored-by: Jacquerel <hnevard@gmail.com>

SethLafuente commented:
> Spiderlings currently get stuck in webs, lol

Yep the person that made spiderlings from structures to mobs didn't make them not get stuck in webs not really my fault but I am trying to fix it

Hoolny committed:
Update code/game/objects/effects/spiderwebs.dm

grammar sense fix

Co-authored-by: Jacquerel <hnevard@gmail.com>

Hoolny committed:
Update spiderwebs.dm

giant spiders can now pull on webs idk how this mistake happened

Hoolny committed:
Update code/game/objects/effects/spiderwebs.dm

Jacquerel fix of pulling in webs and stuck

Co-authored-by: Jacquerel <hnevard@gmail.com>

Hoolny committed:
Update giant_spider_variants.dm

sense fix spider ambush comment definition

Hoolny committed:
Update giant_spider_variants.dm

no flavor text for spider code definition understandable

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spider_variants.dm

,

Co-authored-by: Jacquerel <hnevard@gmail.com>

Hoolny committed:
Update giant_spider_variants.dm

tangle spider better description

Hoolny committed:
Update hivemind.dm

fixed a small typo of message description saying command (the BIG text isn't at typo though the command message is still BIG)

Hoolny committed:
Update spiderwebs.dm

woops all 	melee_cooldown_time = 0

Hoolny committed:
Update web.dm

Hoolny committed:
Update spiderwebs.dm

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
Update hivemind.dm

new proc :D

Hoolny committed:
Update hivemind.dm

still need the icon to be different

Hoolny committed:
Update spiderling_subtypes.dm

it looks defenseless

Hoolny committed:
Update spider_roles.dm

faster

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
push push puush fixes

Hoolny committed:
ye

Hoolny committed:
h

Hoolny committed:
charge fix

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
thanks jaqc

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm

ye

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
ye

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
flesh spider tweaks

Hoolny committed:
ye des

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
lovely chaps

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
eg

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
FINALLY

Hoolny committed:
incoorection description

Hoolny committed:
better naming

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

Toastgoats commented:
As cool as this is, I really think you should address the atmos issue with spiders before adding a whole slew of new content for them.
Whats the point of adding a bajillion new spiders and abilities if you still die because you were on the same screen as a 1 tile hullbreach 10 seconds ago.

SethLafuente commented:
> As cool as this is, I really think you should address the atmos issue with spiders before adding a whole slew of new content for them.
> 
> Whats the point of adding a bajillion new spiders and abilities if you still die because you were on the same screen as a 1 tile hullbreach 10 seconds ago.

Oh thank god I thought this was a serious comment just crushing this whole PR but is just someone who dosent know what their talking about not an Insult but don't judge before you play...

Anyways first of all the addition of this spiders isn't adding an issue "which it's not my responsibility to fix" so to begin with this makes no sense 

If you had actually looked and played this new features you would know I added a whole collection of webs to stop this from happening which are pretty sweet that being web passage which stops atmos but let's spider through not to mention this issue was fixed with sealed webs 

This wasent only me going uuuh more spiders which if it was that would have being fine but no this was me balancing spiders and adding plenty of difference in mechanics to balance many issues players saw which include the atmos issue so sleep sound knowing that not only do you get cool new spiders but you get cool new features to fix existing balance issues ❤️

Toastgoats commented:
> > As cool as this is, I really think you should address the atmos issue with spiders before adding a whole slew of new content for them.
> > Whats the point of adding a bajillion new spiders and abilities if you still die because you were on the same screen as a 1 tile hullbreach 10 seconds ago.
> 
> Oh thank god I thought this was a serious comment just crushing this whole PR but is just someone who dosent know what their talking about not an Insult but don't judge before you play...
> 
> Anyways first of all the addition of this spiders isn't adding an issue "which it's not my responsibility to fix" so to begin with this makes no sense
> 
> If you had actually looked and played this new features you would know I added a whole collection of webs to stop this from happening which are pretty sweet that being web passage which stops atmos but let's spider through not to mention this issue was fixed with sealed webs
> 
> This wasent only me going uuuh more spiders which if it was that would have being fine but no this was me balancing spiders and adding plenty of difference in mechanics to balance many issues players saw which include the atmos issue so sleep sound knowing that not only do you get cool new spiders but you get cool new features to fix existing balance issues ❤️

No problem, just didn't see any mention of that issue in the hackmd.

SethLafuente commented:
I didn't see a need to address it because sealed webs already fix it but it was under tangle spider sbilities and the general theme of making spiders a more defend the castle kind of antag the issue you mentioned just now isn't one anymore but I made the issue even more irrelevant since I added those webs but I'll make sure to add this portion to the hack MD

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Hunter spider slight health buff

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
guard spider carcass

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
correct

Hoolny committed:
Update hivemind.dm

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
woops

Hoolny committed:
woops

eg

Hoolny committed:
sprite color better

Hoolny committed:
Update giant_spider_variants.dm 45 obj damage

Hoolny committed:
Merge branch 'master' into Spider-Scout

MrMelbert commented on code/datums/actions/mobs/sneak.dm:
```diff
@@ -0,0 +1,32 @@
+
+	return ..()
+
+/datum/action/cooldown/sneak/spider/Activate(atom/target)
```
```suggestion
/datum/action/cooldown/sneak/Activate(atom/target)
```

MrMelbert commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web_sneak.dm:
Don't need any of this code anymore

MrMelbert commented on code/modules/mob/living/carbon/alien/adult/alien_powers.dm:
```diff
@@ -338,32 +338,31 @@ Doesn't work on other aliens/AI.*/
-/datum/action/cooldown/alien/sneak/Remove(mob/living/remove_from)
-	if(HAS_TRAIT(remove_from, TRAIT_ALIEN_SNEAK))
+/datum/action/cooldown/sneak/alien/Remove(mob/living/remove_from)
+	if(HAS_TRAIT(remove_from, TRAIT_SNEAK))
 		remove_from.alpha = initial(remove_from.alpha)
-		REMOVE_TRAIT(remove_from, TRAIT_ALIEN_SNEAK, name)
+		REMOVE_TRAIT(remove_from, TRAIT_SNEAK, name)
```
Don't need any of this code anymore

MrMelbert commented on code/modules/mob/living/carbon/alien/adult/alien_powers.dm:
```diff
@@ -338,32 +338,31 @@ Doesn't work on other aliens/AI.*/
 	name = "Sneak"
 	desc = "Blend into the shadows to stalk your prey."
-	button_icon_state = "alien_sneak"
+	button_icon_state = "sneak_alien"
```
This seems like a mistake, don't change the icon state of this unless you're changing it in the `dmi` file itself

SethLafuente replied:
yeah big mistake

MrMelbert commented on code/modules/mob/living/carbon/alien/adult/alien_powers.dm:
```diff
@@ -338,32 +338,31 @@ Doesn't work on other aliens/AI.*/
 	return TRUE
 
-/datum/action/cooldown/alien/sneak
+/datum/action/cooldown/sneak/alien
```
Make sure this stuff caries over from the old imlpementation
`panel = "Alien"`
`check_flags = AB_CHECK_IMMOBILE | AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED`

Hoolny committed:
Update code/datums/actions/mobs/sneak.dm

forgot to remove spider

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
Update web_sneak.dm

removes unnecessary code from web_sneak

Hoolny committed:
Update alien_powers.dm

changed alien sneak to correct

Hoolny committed:
Update alien_powers.dm

maybe its wrong

Hoolny committed:
yes

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
ye

Hoolny committed:
ye

Hoolny committed:
Merge remote-tracking branch 'upstream/master' into Spider-Scout

Hoolny committed:
alien_sneak

Hoolny committed:
ye

Hoolny committed:
reverse reverse

Hoolny committed:
reverse back

Hoolny committed:
Merge branch 'master' into Spider-Scout

MrMelbert commented on code/datums/actions/mobs/charge.dm:
```diff
@@ -177,6 +177,7 @@
 	var/shake_duration = 1 SECONDS
 	/// Intensity of shaking animation
 	var/shake_pixel_shift = 2
+	melee_cooldown_time = 0
```
move this above new var declarations

SethLafuente replied:
donde did it

MrMelbert commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm:
```diff
@@ -61,10 +61,28 @@
+ * Formats the string to have an appropiate size and text color
+ */
+/datum/action/command_spiders/proc/format_message(mob/living/user, message)
+	return span_spiderbroodmother(("<b>Command from [user]:</b> [message]"))
```
```suggestion
	return span_spiderbroodmother("<b>Command from [user]:</b> [message]")
```

MrMelbert commented on code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm:
```diff
@@ -10,6 +10,7 @@
 	cooldown_time = 0 SECONDS
 	/// How long it takes to lay a web
 	var/webbing_time = 4 SECONDS
+	melee_cooldown_time = 0
```
move this above new var declarations

SethLafuente replied:
yes

MrMelbert reviewed (approved):
Small nitpicks, anyone is free to merge after resolved (also pending sprite approval too I guess)

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Update code/modules/mob/living/basic/space_fauna/spider/spider_abilities/hivemind.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Hoolny committed:
Update charge.dm

Hoolny committed:
Update web.dm

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Merge branch 'master' into Spider-Scout

SethLafuente commented:
if this is godo to get merged I'm fine with it been merged

Hoolny committed:
Update spiderwebs.dm

Hoolny committed:
ORANGES WONT KILL THE SPIDERS

Hoolny committed:
Merge branch 'Spider-Scout' of https://github.com/SethLafuente/tgstation into Spider-Scout

Hoolny committed:
Update spiderwebs.dm

Hoolny committed:
back reverse back

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Update giant_spider_variants.dm

Hoolny committed:
Merge branch 'master' into Spider-Scout

ZephyrTFA commented on code/game/objects/effects/spiderwebs.dm:
```diff
@@ -132,3 +164,41 @@
+
+/obj/structure/spider/spikes/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/caltrop, min_damage = 20, max_damage = 30, flags = CALTROP_NOSTUN | CALTROP_BYPASS_SHOES)
```
this is way way too much damage

SethLafuente replied:
tested it for around 200 rounds it is not

SethLafuente replied:
each limb takes up to only 50 damage and if each leg has 50 each it doesn't crit

SethLafuente replied:
making it any less would make the spikes trivial

SethLafuente replied:
> this is way way too much damage

Also is the exact same damage as punji bamboo spikes BUT it doesn't insta perma stun which essentially makes this a weaker version of something crewmembers can make with a 40 second cooldown

SethLafuente replied:
![image](https://github.com/tgstation/tgstation/assets/84478872/6f86c4b8-6ee8-4067-8d8d-b39885761292)

Hoolny committed:
Merge branch 'master' into Spider-Scout

Hoolny committed:
Update giant_spider_variants.dm 65

Hoolny committed:
Merge branch 'master' into Spider-Scout
